test/per_file_coverage.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# directory:coverage_percent
+# for existing extensions with low coverage.
+declare -a KNOWN_LOW_COVERAGE=(
+"source/extensions/common:95.1"
+"source/extensions/common/crypto:91.5"
+"source/extensions/common/wasm:87.8"
+"source/extensions/common/wasm/v8:88.3"
+"source/extensions/common/wasm/null:77.8"
+"source/extensions/compression:90.1"
+"source/extensions/compression/gzip:93.1"
+"source/extensions/compression/gzip/decompressor:81.1"
+"source/extensions/compression/common:60.0"
+"source/extensions/compression/common/decompressor:20.0"
+"source/extensions/filters/network/sni_cluster:90.3"
+"source/extensions/filters/network/thrift_proxy/docs:0"
+"source/extensions/filters/network/thrift_proxy/router:96.4"
+"source/extensions/filters/network/sni_dynamic_forward_proxy:92.4"
+"source/extensions/filters/network/dubbo_proxy:96.7"
+"source/extensions/filters/network/dubbo_proxy/router:96.1"
+"source/extensions/filters/network/direct_response:89.3"
+"source/extensions/filters/network/kafka/serialization:0"
+"source/extensions/filters/network/kafka/protocol:0"
+"source/extensions/filters/http/dynamic_forward_proxy:93.2"
+"source/extensions/filters/http/cache:80.8"
+"source/extensions/filters/http/cache/simple_http_cache:84.5"
+"source/extensions/filters/http/csrf:96.6"
+"source/extensions/filters/http/ip_tagging:92.0"
+"source/extensions/filters/http/decompressor:80"
+"source/extensions/filters/http/compressor:84.4"
+"source/extensions/filters/http/header_to_metadata:95.0"
+"source/extensions/filters/http/grpc_json_transcoder:93.3"
+"source/extensions/filters/http/aws_request_signing:93.3"
+"source/extensions/filters/listener:95.7"
+"source/extensions/filters/listener/tls_inspector:92.9"
+"source/extensions/filters/listener/http_inspector:93.3"
+"source/extensions/filters/udp:91.2"
+"source/extensions/filters/udp/dns_filter:84.1"
+"source/extensions/filters/common:96.4"
+"source/extensions/filters/common/expr:92.2"
+"source/extensions/filters/common/rbac:93.0"
+"source/extensions/grpc_credentials:93.9"
+"source/extensions/grpc_credentials/aws_iam:88.6"
+"source/extensions/quic_listeners:85.1"
+"source/extensions/quic_listeners/quiche:85.1"
+"source/extensions/quic_listeners/quiche/platform:0"
```

Comment:
You don't need those coverage 0 directories because they ended up into 'n'?

Reply:
Sure  Leaving QUIC in though, as there's code there and I think we should go through and fix extensions in this list eventually.

Reply:
SG